packages/repository/src/__tests__/unit/decorator/model-and-relation.decorator.unit.ts

```diff
@@ -132,6 +132,24 @@ describe('model decorator', () => {
     recentOrders?: Order[];
   }
 
+  it('hides a property defined as hidden', () => {
+    @model()
+    class Client extends Entity {
+      @property()
+      name: string;
+      @property({hidden: true})
+      password: string;
+
+      constructor(data: Partial<Client>) {
+        super(data);
+      }
+    }
+
+    expect(Client.definition.settings.hiddenProperties).to.containEql(
+      'password',
+    );
+  });
+
```

Comment:
We should keep your first assertion too, which checked the `password` property is not present in the instance of `Client`, to make sure the intended functionality actually works.